### PR TITLE
Main logic (mainly) complete, will run when a token is inputted

### PR DESCRIPTION
**Pull request by vDokkaebi**
No events have been bound, no commands registered, no command code yet

Just needs a token in Configuration/Configuration.json, and a bot avatar to be inputted to the json to run.

### CODE REVIEW

File: RedPandaBot.CLI/Bot.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        public SlashCommandsExtension? SlashCommands { get; private set; }
+        public static ConfigJson Config = new ConfigJson();
+        public static readonly EventId BotEventId = new EventId(1);
+
+        public static void Main(string[] args)
+        {
+            var bot = new RedPandaBot();
+            try
+            {
+                bot.RunAsync().GetAwaiter().GetResult();
```
Comment (Petrosz007) [inline RedPandaBot.CLI/Bot.cs:16]:
```suggestion
        public static async Task Main(string[] args)
        {
            var bot = new RedPandaBot();
            try
            {
                await bot.RunAsync();
```

File: RedPandaBot.CLI/Bot.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            Log.Information("Connecting to Discord..");
+            await ShardedClient.StartAsync().ConfigureAwait(false);
+            Log.Information("Connected to Discord successfully!");
+
+            Log.Information($"Bot version {Assembly.GetExecutingAssembly().GetName().Version}");
+
+            await Task.Delay(-1).ConfigureAwait(false);
+        }
+    }
+}
```
Comment (Petrosz007) [inline RedPandaBot.CLI/Bot.cs:85]:
```suggestion
}

```

File: RedPandaBot.CLI/Bot.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            });
+
+            var slashCommandsConfig = await ShardedClient.UseSlashCommandsAsync();
+
+            // Command registration will go here
+
+            // Client, Slash Command and error event binding will happen here
+
+            Log.Information("Connecting to Discord..");
+            await ShardedClient.StartAsync().ConfigureAwait(false);
```
Comment (Petrosz007) [inline RedPandaBot.CLI/Bot.cs:77]: If we aren't using `GetAwaiter().GetResult()` or `Result` is `ConfigureAwait(false)` required here?

File: RedPandaBot.CLI/GlobalUsings.cs [hunk 4]
```diff
@@ -1 +1,22 @@
...
+global using DSharpPlus.SlashCommands;
+global using DSharpPlus.SlashCommands.Attributes;
+global using DSharpPlus.SlashCommands.EventArgs;
+global using Newtonsoft.Json;
+global using Serilog;
+global using Serilog.Events;
+global using Serilog.Extensions.Logging;
+global using Serilog.Sinks.SystemConsole.Themes;
+global using Microsoft.Extensions.Logging;
+global using RedPandaBot.Configuration;
```
Comment (Petrosz007) [inline RedPandaBot.CLI/GlobalUsings.cs:22]:
```suggestion
global using RedPandaBot.Configuration;

```

File: RedPandaBot.CLI/Bot.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
+﻿namespace RedPandaBot
+{
+    public class RedPandaBot
+    {
+        public DiscordShardedClient? ShardedClient { get; private set; }
+        public InteractivityExtension? Interactivity { get; private set; }
+        public SlashCommandsExtension? SlashCommands { get; private set; }
+        public static ConfigJson Config = new ConfigJson();
+        public static readonly EventId BotEventId = new EventId(1);
```
Comment (Petrosz007) [inline RedPandaBot.CLI/Bot.cs:9]:
```suggestion
        private DiscordShardedClient? ShardedClient { get; private set; }
        private InteractivityExtension? Interactivity { get; private set; }
        private SlashCommandsExtension? SlashCommands { get; private set; }
        private static ConfigJson Config = new ConfigJson();
        private static readonly EventId BotEventId = new EventId(1);
```

File: RedPandaBot.CLI/Bot.cs [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+        public static void Main(string[] args)
```
Comment (Petrosz007) [inline RedPandaBot.CLI/Bot.cs:11]: I think, that `Main` shouldn't exist inside the `RedPandaBot` class, it should be in `Program.cs`, to separate the bot logic from the actual program running.

File: RedPandaBot.CLI/Bot.cs [hunk 7]
```diff
@@ -0,0 +1,85 @@
+﻿namespace RedPandaBot
```
Comment (Petrosz007) [inline RedPandaBot.CLI/Bot.cs:1]:
```suggestion
﻿namespace RedPandaBot.CLI
```

File: RedPandaBot.CLI/Bot.cs [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+{
+    public class RedPandaBot
```
Comment (Petrosz007) [inline RedPandaBot.CLI/Bot.cs:3]: If the class is called `RedPandaBot` the filename should be `RedPandaBot.cs`

File: RedPandaBot.CLI/Configuration/ConfigJson.cs [hunk 9]
```diff
@@ -0,0 +1,14 @@
+﻿namespace RedPandaBot.Configuration
```
Comment (Petrosz007) [inline RedPandaBot.CLI/Configuration/ConfigJson.cs:1]:
```suggestion
﻿namespace RedPandaBot.CLI.Configuration
```

**Review by Petrosz007 — CHANGES_REQUESTED** [review]
Nice work! Sorry for nitpicking, I tried to add the small things as suggestions.

File: RedPandaBot.CLI/Bot.cs (hunk 3 above)
Comment (vDokkaebi) [reply RedPandaBot.CLI/Bot.cs:77]: It's recommended to use with ShardedClients, but it's not mandatory.

**Comment by Petrosz007** [comment]
```
error NU1102: Unable to find package DSharpPlus with version (>= 4.2.0-nightly-01018)
```
The CI build can't download DSharpPlus, can we add a NuGet.config somewhere with [this external source](https://nuget.emzi0767.com/gallery/about) or version `4.1.0` is enough and we can [simply use NuGet](https://www.nuget.org/packages/DSharpPlus)?

**Comment by vDokkaebi** [comment]
> ```
> error NU1102: Unable to find package DSharpPlus with version (>= 4.2.0-nightly-01018)
> ```
> 
> The CI build can't download DSharpPlus, can we add a NuGet.config somewhere with [this external source](https://nuget.emzi0767.com/gallery/about) or version `4.1.0` is enough and we can [simply use NuGet](https://www.nuget.org/packages/DSharpPlus)?

I can add the NuGet Config for the nightly build - 4.1.0 isn't a new enough version for SlashCommands.

**Review by Petrosz007 — APPROVED** [review]

**Comment by vDokkaebi** [comment]
@Petrosz007 | What's your Discord username so we can discuss your plans with the bot? I'll be happy to help get it fully up and running. :)

**Comment by Petrosz007** [comment]
We can use the [discussions](https://github.com/Petrosz007/RedPandaBot/discussions) page of the repo. My friends will join the development because of HacktoberFest too, so we should have the discussions open for anyone.

**Comment by vDokkaebi** [comment]
> We can use the [discussions](https://github.com/Petrosz007/RedPandaBot/discussions) page of the repo. My friends will join the development because of HacktoberFest too, so we should have the discussions open for anyone.

Alright sounds good.

**Pull request merged**